src/Storefront/Framework/Twig/TemplateDataExtension.php

```diff
@@ -75,4 +80,36 @@ private function getControllerInfo(Request $request): ControllerInfo
 
         return $controllerInfo;
     }
+
+    private function minSearchLength(SalesChannelContext $context): int
+    {
+        $query = $this->connection->createQueryBuilder();
+
+        $query->select('min_search_length')
+            ->from('product_search_config')
+            ->where('language_id = :id')
+            ->setParameter('id', Uuid::fromHexToBytes($context->getLanguageId()));
+
+        $min = $query->executeQuery()->fetchOne();
+
+        return $min ? (int) $min : 3;
+    }
+
+    private function getLanguage(SalesChannelContext $context): array
```

Comment:
The language informations belong to the SaleschannelContext, as the currencies do.
Also minSearchLength seam to be a bit random to be selected as "always available". Either we have an object that holds bearly changed standard configurations or we get them when we need them, but select single configs one after one to be stored in the twigContext does make things more complex and unreliable, IMHO.